packages/lesswrong/server/editor/conversionUtils.ts

```diff
@@ -599,6 +601,88 @@ function googleDocFormatting(html: string): string {
   return $.html();
 }
 
+/**
+ * Convert the CSS based "cropping" in the imported html into actual cropping
+ */
+async function googleDocCropImages(html: string): Promise<string> {
+  // Example of CSS-based cropping:
+  // <p>
+  //   <span style="overflow: hidden; display: inline-block; width: 396.00px; height: 322.40px;">
+  //     <img src="https://example.com/image.jpg"
+  //          style="width: 602.00px; height: 427.97px; margin-left: -110.00px; margin-top: -48.60px;">
+  //   </span>
+  // <p>
+
+  const $ = cheerio.load(html);
+
+  const cropPromises = $('p > span:has(> img)').map(async (_, span) => {
+    const $span = $(span);
+    const $img = $span.find('img');
+
+    if ($img.length === 0) return;
+
+    const spanStyle = $span.attr('style');
+    const imgStyle = $img.attr('style');
+    const src = $img.attr('src');
+
+    if (!spanStyle || !imgStyle || !src) return;
+
+    const spanWidth = parseFloat(spanStyle.match(/width:\s*([\d.]+)px/)?.[1] || '0');
+    const spanHeight = parseFloat(spanStyle.match(/height:\s*([\d.]+)px/)?.[1] || '0');
+    const imgWidth = parseFloat(imgStyle.match(/width:\s*([\d.]+)px/)?.[1] || '0');
+    const imgHeight = parseFloat(imgStyle.match(/height:\s*([\d.]+)px/)?.[1] || '0');
+    const marginLeft = parseFloat(imgStyle.match(/margin-left:\s*([-\d.]+)px/)?.[1] || '0');
+    const marginTop = parseFloat(imgStyle.match(/margin-top:\s*([-\d.]+)px/)?.[1] || '0');
+
+    const leftRelative = Math.max(-marginLeft, 0) / imgWidth;
+    const topRelative = Math.max(-marginTop, 0) / imgHeight;
+    const widthRelative = Math.round(spanWidth) / imgWidth;
+    const heightRelative = Math.round(spanHeight) / imgHeight;
+
+    if (leftRelative === 0 && topRelative === 0 && widthRelative === 1 && heightRelative === 1) {
+      return
+    }
+
+    try {
+      const response = await axios.get(src, { responseType: 'arraybuffer' });
+      const buffer = Buffer.from(response.data, 'binary');
+
+      const image = await Jimp.read(buffer);
```

Comment:
I started off using [sharp](https://sharp.pixelplumbing.com/) but it wouldn't work in CI due a problem with supporting the linux-x64 architecture that can only be fixed by upgrading our version of yarn to 3+. That seemed like overkill, so I switched to [Jimp](https://github.com/jimp-dev/jimp) which is apparently quite slow by comparison (see [here](https://sharp.pixelplumbing.com/performance)). It still seemed reasonably fast with about ~15 cropped images in one post though so I don't expect this to be a problem